roles/openshift_on_openstack/templates/prepare_block_devices.yml

```diff
@@ -18,6 +19,25 @@
         path: "{{ mount_path }}"
         state: directory
 
+    # Get etcd storage device state.
+    - name: Get etcd storage device state
+      command: "pvdisplay -C --noheadings -o pv_name,vg_name {{ device_path }}"
+      register: device_info
+      failed_when: False
+      when: etcd_device_wipe
+
+    # Clear etcd storage device contents.
+    - name: Clear etcd storage device contents
+      shell: "{% for line in device_info.stdout_lines %}{% set fields = line.split() %}{% if fields | count > 1 %}lvremove -ff {{ fields[1] }}; vgremove -fy {{ fields[1] }}; {% endif %}pvremove -fy {{ fields[0] }}; {% endfor %}"
```

Comment:
I see you are using Jinja syntax here, but checking the code this file is only copied and not rendered by the template Ansible module. So I do not think this Jinja would get processed as you are intending.

If we are to use "template" module all the other curly brackets ({) in the file would need to be escaped. Could we add an Ansible task to compute the fields and just use the resulting variable?

Reply:
Well, @mbruzek, I very rarely submit PRs without actually testing them: http://pastebin.test.redhat.com/556003

Tested both on a KVM and the old OpenStack install.  Please re-consider.